sklearn/utils/testing.py

```diff
@@ -398,6 +407,188 @@ def assert_raise_message(exception, message, function, *args, **kwargs):
         assert_in(message, error_message)
 
 
+def assert_array_not_equal(x, y, err_msg="The arrays are equal", verbose=True):
+    """Raises an AssertionError if two array_like objects are equal."""
+    try:
+        assert_array_equal(x, y, verbose=verbose)
+    except AssertionError:
+        assert True
+        return
+
+    assert False, err_msg
+
+
+def assert_not_raises(given_exception, function, *args, **kwargs):
```

Comment:
There should certainly be a `raise` for any other exception types, but I agree, this is a bit obscure given that the statement itself will only succeed if the exception is not raised. Arguably, this helper adds some legibility.